controllers/openstackdataplane_controller.go

```diff
@@ -207,34 +220,25 @@ func (r *OpenStackDataPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 		r.Log.Info("one or more roles aren't ready, requeueing")
 		return ctrl.Result{RequeueAfter: time.Second * 5}, nil
 	}
-	if instance.Spec.DeployStrategy.Deploy && len(deployErrors) == 0 {
+	if len(deployErrors) == 0 {
 		instance.Status.Deployed = true
 		r.Log.Info("Set DeploymentReadyCondition true", "instance", instance)
 		instance.Status.Conditions.Set(condition.TrueCondition(condition.DeploymentReadyCondition, condition.DeploymentReadyMessage))
 	}
 
-	// Set DeploymentReadyCondition to False if it was unknown.
-	// Handles the case where the DataPlane is created with
-	// DeployStrategy.Deploy=false.
-	if instance.Status.Conditions.IsUnknown(condition.DeploymentReadyCondition) {
-		r.Log.Info("Set DeploymentReadyCondition false")
-		instance.Status.Conditions.Set(condition.FalseCondition(condition.DeploymentReadyCondition, condition.NotRequestedReason, condition.SeverityInfo, condition.DeploymentReadyInitMessage))
-	}
-
-	// Explicitly set instance.Spec.Deploy = false
-	// We don't want another deploy triggered by any reconcile request, it should
-	// only be triggered when the user (or another controller) specifically
-	// sets it to true.
-	instance.Spec.DeployStrategy.Deploy = false
-
 	// Set DeploymentReadyCondition to False if it was unknown.
 	// Handles the case where the Node is created with
-	// DeployStrategy.Deploy=false.
 	if instance.Status.Conditions.IsUnknown(condition.DeploymentReadyCondition) {
 		r.Log.Info("Set DeploymentReadyCondition false")
 		instance.Status.Conditions.Set(condition.FalseCondition(condition.DeploymentReadyCondition, condition.NotRequestedReason, condition.SeverityInfo, condition.DeploymentReadyInitMessage))
 	}
 
+	// We don't want another deploy triggered by any reconcile request, it
+	// should only be triggered when the user (or another controller)
+	// specifically clears this annotation and requests it.
+	r.Log.Info("Setting deploy protection annotation")
+	instance.ObjectMeta.SetAnnotations(redeployProtectionAnnotation)
```

Comment:
> the reconcile is requeued due to the failure and it will not see the annotation so ti will re-run the ansible tasks

Do we link the "successful deployment" from the step 1 with the object containing that deploy-protection annotation? If we do, the stored hash in its status should include its linked AEE job status, if there is a link to an existing job. Then the re-queued reconciler will notice that the re-calculated hash matches the linked AEE status and will re-add the lost deploy-protection annotation.  But if we delete the linked job, then ansible will become re-executed

Reply:
> > the reconcile is requeued due to the failure and it will not see the annotation so ti will re-run the ansible tasks
> 
> Do we link the "successful deployment" from the step 1 with the object containing that deploy-protection annotation? If we do, the stored hash in its status should include its linked AEE/job status, if there is a link to an existing one. Then the re-queued reconciler will notice that the re-calculated hash matches the linked AEE status and will re-add the lost deploy-protection annotation, while skipping ansible execution. But if we delete the linked job, then ansible would be executed again

This is what we do with the Jobs. i) we store the hash and ii) we don't delete the job long enough that we can make sure that i) is succeeded.